app/data_model/list_store.py

```diff
@@ -12,22 +13,92 @@ def random_string(length):
     return ''.join(random.choice(ascii_letters) for _ in range(length))
 
 
+class ListRepresentation:
+    def __init__(
+        self, items: Optional[List] = None, primary_person: Optional[str] = None
+    ):
+        self.items = items or []
+        self.primary_person = primary_person
+
+    def __eq__(self, other):
+        if not isinstance(other, ListRepresentation):
+            return NotImplemented
+        return self.items == other.items and self.primary_person == other.primary_person
+
+    def remove_item_id(self, to_remove):
+        """ Remove an item from the list.
+        If the list item does not exist, this will not raise an error.
+        """
+        try:
+            self.items.remove(to_remove)
+        except ValueError:
+            pass
+
+        if self.primary_person == to_remove:
+            self.primary_person = None
+
+    @classmethod
+    def from_dict(cls, input_dict: dict):
+        return cls(
+            items=input_dict['items'], primary_person=input_dict.get('primary_person')
+        )
+
+    def to_dict(self, name=None):
+        serialised = {'items': self.items}
+        if self.primary_person:
+            serialised['primary_person'] = self.primary_person
+
+        if name:
+            serialised['name'] = name
+
+        return serialised
+
+    def __repr__(self):
+        return f'<ListRepresentation items={self.items}, primary_person={self.primary_person}>'
+
+
 class ListStore:
     """ Storage for lists.
+
+    Stores in the form:
+
+    ```
+    self._lists: {
+        list_name_1: ListRepresentation(
+            items: [1,2,3],
+            primary_person: 1
+        )
+    }
+    ```
+
+    serialises to:
+
+    ```
+    [
+        {
+            name: list_name_1,
+            primary_person: 1
+            items: [1,2,3]
+        }
+    ]
+    ```
     """
 
-    def __init__(self, existing_items=None):
-        if not existing_items:
-            existing_items = []
+    def __init__(self, existing_items: Mapping[str, ListRepresentation] = None):
```

Comment:
Is this type hint accurate? I think it should be `Union[Mapping[str, ListRepresentation], None]`